### ISSUE TITLE
node-compat: process.argv0

### ISSUE THREAD

**Issue by chibat** [denoland/deno#20934]
The result when explicitly specifying the process name is different from Node.js.
I did a test with a canary release.

Node
```
$ bash -c 'exec -a customArgv0 node'
> process.argv0
'customArgv0'
```

Deno
```
$ deno --version
deno 1.37.2+5095af7 (canary, x86_64-unknown-linux-gnu)
v8 11.8.172.13
typescript 5.2.2
$ bash -c 'exec -a customArgv0 deno'
> import process from "node:process"
undefined
> process.argv0
"deno"
```

related issue: #20924 
related PR: #20925

**Comment by birkskyum** [comment]
Closed by
- https://github.com/denoland/deno/pull/20925

**Comment by chibat** [comment]
The fix in PR #20924 does not produce the same results as Node.js.

**Comment by satyarohith** [comment]
This is still an issue with the latest version of deno:
```
➜  ~ bash -c 'exec -a customArgv0 node'
Welcome to Node.js v21.5.0.
Type ".help" for more information.
> process.argv0
'customArgv0'
> .exit
➜  ~ bash -c 'exec -a customArgv0 deno'
Deno 1.40.5
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> import process from "node:process"
undefined
> process.argv0
"deno"
>
```
